compiler/resolution/functionResolution.cpp

```diff
@@ -8361,7 +8364,7 @@ static void insertReturnTemps() {
       if (call->list == NULL && isForallRecIterHelper(call))
         continue;
       if (FnSymbol* fn = call->resolvedOrVirtualFunction()) {
-        if (fn->retType != dtVoid) {
+        if (fn->retType != dtVoid && fn->retTag != RET_TYPE) {
```

Comment:
Should this code run for fns returning a runtime type?

Reply:
Yes, but in such cases ``fn->retTag`` is ``RET_VALUE``.